safe/gui/tools/batch/batch_dialog.py

```diff
@@ -402,7 +402,7 @@ def define_layer(self, layer_path):
             else:
                 LOGGER.critical('Failed to create layer from scenario')
                 return
-        elif extension in ['.shp', '.geojson', '.gpkg']:
+        elif extension in ['.shp', '.json', '.geojson', '.gpkg']:
```

Comment:
ok this is just a bandaid fix for now ... we really should support all QGIS supported formats. We could try to load each layer first as a raster, then as a vector then fail.